mergekit/plan.py

```diff
@@ -85,16 +86,46 @@ def __init__(
         if config.base_model and config.align_weights:
             self._space_planner = SpacePlanner(config.base_model)
 
-    @lru_cache
-    def model_arch_info(self, model: ModelReference):
-        return ConfiguredArchitectureInfo(
-            info=self.arch_info,
-            config=model.config(trust_remote_code=self.trust_remote_code),
-        )
+        self.arch_dict = {}
+        _models = config.referenced_models()
+        base_model = _models[0]
+        base_config = base_model.config(trust_remote_code=options.trust_remote_code)
+        self.arch_info = get_architecture_info(base_config)
+
+        for m in config.referenced_models():
+            m_config = m.config(trust_remote_code=options.trust_remote_code)
+            configured_arch_info = ConfiguredArchitectureInfo(
+                info=get_architecture_info(m_config),
+                config=m.config(m_config),
+            )
 
-    def normalize_config(self):
-        base_model = self.config.base_model
+            if config.align_weights:
+                mapping = None
+                is_same_arch = False
+
+                for arch, destination_arch in [
+                    (arch1, arch2)
+                    for arch1 in base_config.architectures
+                    for arch2 in m_config.architectures
+                ]:
+                    if arch == destination_arch:
+                        is_same_arch = True
+                        break
```

Comment:
I think this logic isn't quite right. Consider a merge with two models (aside from the base) - the first is the same architecture as the base and the second is a different architecture. This will see that the first model has the same architecture and break out of the loop, so the second will not get overrides set.

Reply:
Perhaps I'm missing something but the inner most loop loops through the alternate names of both architectures and breaks if it's found a match
But the outermost loop takes care of the actual model list  traversal

unless `break` breaks out of both the loops  which I don't think is the case, this should be fine. It is possible I'm overlooking something very trivial

Reply:
Oh, you're absolutely right, sorry about that - I missed the outer loop. Disregard!

Reply:
I wonder if I should name things better?